lib/rules/no-hooks-from-ancestor-modules.js

```diff
@@ -59,11 +59,10 @@ module.exports = {
                 isInModuleCallbackBody(node);
         }
 
-        function isFunctionWithIdentifierArg(node) {
-            return node &&
-                node.type === "FunctionExpression" &&
-                node.params.length > 0 &&
-                node.params[0].type === "Identifier";
+        function getNonThisIdentifierFromParams(params) {
+            // In TypeScript, `this` can be passed as the first function parameter to add a type to it,
+            // and we want to ignore that parameter since we're looking for the `hooks` variable.
+            return params.find((param, i) => param.type === "Identifier" && !(i === 0 && param.name === "this"));
```

Comment:
this will work because we expect only identifier params in these callbacks, but in theory this could allow returning a param other than the first or second

this also makes me wonder if destructuring hooks works

Reply:
Good point, corrected logic.